Add more questions to the application and also some improvements from the last version

chacaa opened the pull request:
I made the app bigger by adding more questions and the possibility to navigate between them (now you can go forward and also backward). It was also added to the app the possibility of changing to the next question only by tapping on the question itself.

### Related Trello cards:
* [[AP - Chapter 2] Android and Model-View-Controller](https://trello.com/c/wOohgOQm)
* [[AP - Chapter 3] The Activity Lifecycle](https://trello.com/c/t3W9RdOZ)

### Additional Info:
* I also added a toast message when you reach the last question and try to continue with the next one tapping on the textview that shows the question.
* Friday 01/27/17 : I Added images for the prev and next button.

### UI preview:
* Portrait mode:
![alt text](http://g.recordit.co/tn1dkE9FkW.gif)

* Landscape mode:
![alt text](http://g.recordit.co/uY48KSfoD2.gif)

/cc @xmartlabs/android

Commit:
i fixed the format of the header

mirland commented on app/src/main/java/com/xmartlabs/scasas/geoquiz/module/Question.java:
```diff
@@ -0,0 +1,46 @@
+/**
+ * Created by scasas on 1/25/17.
+ */
+
```
remove space

diegomedina248 replied:
If you click on the "light bulb" you can edit the template and remove that space there so you don't have to do it manually

mirland commented on app/src/main/java/com/xmartlabs/scasas/geoquiz/module/Question.java:
```diff
@@ -0,0 +1,46 @@
+    textResId = builder.textResId;
+    answerTrue = builder.answerTrue;
+  }
+
```
remove space

mirland commented on app/src/main/java/com/xmartlabs/scasas/geoquiz/ui/QuizActivity.java:
```diff
@@ -17,12 +19,38 @@
   private Button trueButon;
+  private Button nextButton;
+  private TextView questionTextView;
+  private Question[] questionBank = new Question[] {
```
We should avoid using Arrays, use a list instead of it

mirland replied:
you can use `Arrays.asList`

mirland replied:
Also this could grow up very quickly, maybe you should put it in a method

diegomedina248 replied:
I agree, also to reduce the duplicated code

mirland commented on app/src/main/java/com/xmartlabs/scasas/geoquiz/ui/QuizActivity.java:
```diff
@@ -46,15 +74,41 @@ private void setupQuizButton() {
+    questionTextView.setText(question);
   }
+
+  private int checkAnswer(String valueButton) {
```
the name is not right, "Check answer" is not what the function does

mirland replied:
also annotate it with the `@StringRes` annotation

mirland commented on app/src/main/java/com/xmartlabs/scasas/geoquiz/ui/QuizActivity.java:
```diff
@@ -46,15 +74,41 @@ private void setupQuizButton() {
   }
+
+  private int checkAnswer(String valueButton) {
+    if (valueButton.toLowerCase().equals(String.valueOf(questionBank[currentIndex].isAnswerTrue()))) {
```
use the shortest form of return

mirland replied:
http://stackoverflow.com/questions/8898590/short-form-for-java-if-statement

mirland commented on app/src/main/java/com/xmartlabs/scasas/geoquiz/module/Question.java:
```diff
@@ -0,0 +1,46 @@
+ */
+
+public class Question {
+  private int textResId;
```
add `@StringRes` annotation

mirland commented on app/src/main/java/com/xmartlabs/scasas/geoquiz/ui/QuizActivity.java:
```diff
@@ -46,15 +74,41 @@ private void setupQuizButton() {
+      @Override
+      public void onClick(View v) {
+        if (currentIndex == questionBank.length - 1) {
+          Toast.makeText(QuizActivity.this, "No more questions for today :)", Toast.LENGTH_SHORT).show();
```
:+1:

diegomedina248 replied:
The string should be made a resource

mirland replied:
yes! my bad!

mirland commented on app/src/main/res/layout/activity_quiz.xml:
```diff
@@ -8,10 +8,10 @@
         android:layout_width="wrap_content"
         android:layout_height="wrap_content"
         android:padding="24dp"
-        android:text="@string/question_text"
```
you shouldn't remove this line, use the tools namespace, `tools:text="@string/question_text"`. The main reason of this is that the layout will be fine in the preview

mirland commented on app/src/main/java/com/xmartlabs/scasas/geoquiz/module/Question.java:
```diff
@@ -0,0 +1,46 @@
+  private int textResId;
+  private boolean answerTrue;
+
+  public int getTextResId() {
```
add `@StringRes` annotation

bryant1410 commented:
I recommend following [a format in git commit messages and PRs titles](http://chris.beams.io/posts/git-commit/). I think [Pro Git](https://git-scm.com/book/en/v2) also has recommendations.

diegomedina248 commented on app/src/main/java/com/xmartlabs/scasas/geoquiz/module/Question.java:
```diff
@@ -0,0 +1,46 @@
+    private int textResId;
+    private boolean answerTrue;
+
+    public Builder() {
```
Is this constructor necessary? It should exist by default

bryant1410 replied:
Just to provide more context, Java creates by default a public non-args constructor to all classes that don't declare a constructor.

chacaa replied:
Yep! I forgot that, thanks!

mirland replied:
BTW: it's autogenerated code :)

diegomedina248 commented on app/src/main/java/com/xmartlabs/scasas/geoquiz/module/Question.java:
```diff
@@ -0,0 +1,46 @@
+    public Builder() {
+    }
+
+    public Builder textResId(int val) {
```
Prefer to use a more descriptive name for variables whenever possible. I'd use textResId here

diegomedina248 commented on app/src/main/java/com/xmartlabs/scasas/geoquiz/module/Question.java:
```diff
@@ -0,0 +1,46 @@
+      return this;
+    }
+
+    public Builder answerTrue(boolean val) {
```
here too

diegomedina248 commented on app/src/main/java/com/xmartlabs/scasas/geoquiz/ui/QuizActivity.java:
```diff
@@ -46,15 +74,41 @@ private void setupQuizButton() {
+    nextButton.setOnClickListener(new View.OnClickListener() {
+      @Override
+      public void onClick(View v) {
+        if (currentIndex == questionBank.length - 1) {
```
just for clarity purposes, consider in these cases extracting this into a new method.
```java
if (isLastQuestion()) {
```
is far more readable.
Just a thought

chacaa replied:
I'll keep it in mind for future commits, thanks!

diegomedina248 reviewed (commented):
Left some comments

Commit:
Changes requested and some improvements

mirland commented on app/src/main/java/com/xmartlabs/scasas/geoquiz/ui/QuizActivity.java:
```diff
@@ -20,29 +24,9 @@
-          .answerTrue(true)
-          .build()
-  };
+  private List<Question> questionBank = new ArrayList<>();
```
add `@NonNull`

mirland replied:
avoid putting `Bank` or something like that. Call it as `questions`. Also in some cases you can use `questionList`

mirland replied:
https://github.com/xmartlabs/Android-Style-Guide#field-method-variable-and-parameter-names

chacaa replied:
done!

mirland commented on app/src/main/java/com/xmartlabs/scasas/geoquiz/ui/QuizActivity.java:
```diff
@@ -70,45 +54,162 @@ public boolean onOptionsItemSelected(MenuItem item) {
   private void setupQuizButton() {
-    trueButon = (Button) findViewById(R.id.true_button);
-    trueButon.setOnClickListener(new View.OnClickListener() {
+    addQuestion();
```
what do you think of naming it `setupQuestions` ?

chacaa replied:
I thought in that name but I'm afraid that maybe someone could get confuse because of the other method that I called setupQuestion

mirland commented on app/src/main/java/com/xmartlabs/scasas/geoquiz/ui/QuizActivity.java:
```diff
@@ -70,45 +54,162 @@ public boolean onOptionsItemSelected(MenuItem item) {
+            prevButton.setEnabled(true);
+          }
+          if (isLastQuestion()) {
+            nextButton.setEnabled(false);
```
you can put something like `nextButton.setEnabled(!isLastQuestion())` and remove the first if of the bock

mirland replied:
the same in `prevButton`

chacaa replied:
I'll check it, thanks!

mirland commented on app/src/main/java/com/xmartlabs/scasas/geoquiz/ui/QuizActivity.java:
```diff
@@ -70,45 +54,162 @@ public boolean onOptionsItemSelected(MenuItem item) {
-    }
-    return R.string.incorrect_toast;
+  @StringRes
+  private int obtainAnswerValue(String valueButton) {
```
add `@NonNull` in the parameter

mirland commented on app/src/main/java/com/xmartlabs/scasas/geoquiz/ui/QuizActivity.java:
```diff
@@ -70,45 +54,162 @@ public boolean onOptionsItemSelected(MenuItem item) {
   }
 
+  private void addQuestion() {
+    int currentIndexAux = currentIndex;
```
you should avoid naming variables with aux prefix

chacaa replied:
What do you suggest then? Something like listIndex instead?

mirland commented on app/src/main/java/com/xmartlabs/scasas/geoquiz/ui/QuizActivity.java:
```diff
@@ -70,45 +54,162 @@ public boolean onOptionsItemSelected(MenuItem item) {
 
+  private void addQuestion() {
+    int currentIndexAux = currentIndex;
+    this.questionBank.add(currentIndexAux,
```
you don't need put the index, you can remove `currentIndexAux`

mirland commented on app/src/main/java/com/xmartlabs/scasas/geoquiz/ui/QuizActivity.java:
```diff
@@ -70,45 +54,162 @@ public boolean onOptionsItemSelected(MenuItem item) {
+            .build()
+    );
+    currentIndexAux++;
+
```
remove space

mirland commented on app/src/main/java/com/xmartlabs/scasas/geoquiz/ui/QuizActivity.java:
```diff
@@ -70,45 +54,162 @@ public boolean onOptionsItemSelected(MenuItem item) {
+    );
+    currentIndexAux++;
+
+    this.questionBank.add(currentIndexAux,
```
remove this, it is not necessary

chacaa replied:
Do you mean the blanc space?

mirland commented on app/src/main/java/com/xmartlabs/scasas/geoquiz/ui/QuizActivity.java:
```diff
@@ -70,45 +54,162 @@ public boolean onOptionsItemSelected(MenuItem item) {
+    falseButton.setOnClickListener(new View.OnClickListener() {
+      @Override
+      public void onClick(View v) {
+        Toast.makeText(QuizActivity.this, obtainAnswerValue(falseButton.getText().toString()), Toast.LENGTH_SHORT).show();
```
what do you think of putting this sentence (and all the similar sentences) in a function? something like `private void showToast(@NonNull String message)`

chacaa replied:
That isn't a bad idea at all! Thanks!

chacaa commented:
I forgot to add images to the next and prev buttons. I'll try to add it today if I can.

Commit:
Changes requested and some improvements suggested in the pull request

Commit:
Changes requested and some improvements suggested in the pull request

Commit:
Changes requested and some improvements suggested in the pull request, also I added images for the prev and next buttons

mirland commented on app/src/main/java/com/xmartlabs/scasas/geoquiz/ui/QuizActivity.java:
```diff
@@ -17,12 +25,19 @@
+  private ImageButton prevButton;
+  private TextView questionTextView;
+  @NonNull
+  private List<Question> questionList = new ArrayList<>();
```
it's ok, but in the future use `questions`. ["If you need to name a List of activities for example, just name it activities"]
(https://github.com/xmartlabs/Android-Style-Guide?files=1#field-method-variable-and-parameter-names)

mirland commented on app/src/main/java/com/xmartlabs/scasas/geoquiz/ui/QuizActivity.java:
```diff
@@ -42,19 +57,143 @@ public boolean onOptionsItemSelected(MenuItem item) {
+
+  @StringRes
+  private int obtainAnswerValue(@NonNull String valueButton) {
+    return (valueButton.toLowerCase().equals(String.valueOf(questionList.get(currentIndex).isAnswerTrue()))) ?
```
remove the `()`

mirland replied:
fix indentation too

mirland commented on app/src/main/java/com/xmartlabs/scasas/geoquiz/ui/QuizActivity.java:
```diff
@@ -42,19 +57,143 @@ public boolean onOptionsItemSelected(MenuItem item) {
+
+  private void setupQuestions() {
+    int listIndex = 0;
+    this.questionList.add(listIndex,
```
remove `this`

mirland replied:
remove the index, it's not necessary

Commit:
Changes requested and some improvements suggested in the pull request, also I added a landscape mode

Commit:
Fix bug that appeared when changing from portrait mode to landscape mode

The bug was that when you change between modes it automatically changes the question in wich you were step on for the first one on the list.
